airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -42,41 +63,60 @@ def __init__(self, api, *args, **kwargs):
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         """Iterate over entities"""
 
+    def filter_input_fields(self, fields: Sequence[str] = None):
+        return list(set(fields) - set(self.non_object_fields))
+
 
 class IncrementalStreamAPI(StreamAPI, ABC):
     @property
     @abstractmethod
     def state_pk(self):
         """Name of the field associated with the state"""
 
+    @property
+    @abstractmethod
+    def account_pk(self):
+        """Name of the field associated with the account_id"""
+
     @property
     def state(self):
```

Comment:
Is this correct? It's super hard to understand what this state object is by just looking at this loop and the setter below. 

```suggestion
    def state(self) -> Dict[str, str]: 
    """
    State is a dictionary of the format {"cursorField_accountID" : cursorValue}
    """
```

Reply:
This string concatenation method is confusing. Why not just have the state be a double map with the following schema: 
```
{account_id: {stream: cursor_value}}
```

This way we don't need to have this string concatenation below and the index-based extraction of account ID

Reply:
We cannot implement such schema, since the [`_read_stream`](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-python/base_python/source.py#L101) method in `BaseSource` works in such a way that there must be a stream at the top level.
But we can apply the following type of schema:
`{stream: {account_id: cursor_value}}`

Reply:
^I prefer that one to the current approach :+1:

Reply:
In fact, this is now implemented, only need to remove string concatenation